app/commodities/page.client.tsx

```diff
@@ -1,79 +1,194 @@
 'use client';
 
 import { useState } from 'react';
-import { Box, Flex, Heading } from '@chakra-ui/react';
-import Form, { SubmitProps } from '@/components/commodities/Form';
+import {
+  Box,
+  Center,
+  Heading,
+  VStack,
+  Alert,
+  AlertIcon,
+  Flex,
+  Button,
+  useMediaQuery,
+} from '@chakra-ui/react';
 import useColorMode from '@/app/_hooks/useColorMode';
 import selectColor from '@/app/_hooks/fontColorSelector';
+import { ICommodity, ICommodityFormResponse } from '@/app/_types/commodity';
+import Form, { SubmitProps } from '@/components/commodities/Form';
+import CommodityFormResponse from '@/components/commodities/CommodityFormResponse';
+import CommodityFormResponseMobile from '@/components/commodities/CommodityFormResponseMobile';
+import layoutConfig from '../_config/layout';
 
-interface ReqBody extends Omit<SubmitProps, 'commodityId' | 'system'> {
-  commodityId: string;
-  referenceLocation: {
-    xcoordinate: number;
-    ycoordinate: number;
-    zcoordinate: number;
-  };
-  system?: string;
+interface IPageClientProps {
+  commodities: ICommodity[] | null;
 }
 
-const PageClient = () => {
+const PageClient: React.FC<IPageClientProps> = ({ commodities }) => {
   const [isLoading, setIsLoading] = useState(false);
+  const [isBuying, setIsBuying] = useState(true);
+  const [submitted, setSubmitted] = useState(false);
+  const [fetchError, setFetchError] = useState<Error | null>(null);
+  const [commodityResponse, setCommodityResponse] = useState<
+    ICommodityFormResponse[]
+  >([]);
+
   const { isDark } = useColorMode();
+  const [isLarge] = useMediaQuery('(min-width: 1024px)');
 
-  const handleSubmit = (data: SubmitProps) => {
+  const handleSubmit = async (data: SubmitProps) => {
     setIsLoading(true);
-    const formatString = (string: string) =>
-      string.split(' ').join('_').toLowerCase();
+    setSubmitted(false);
+
+    if (data.minDemand === 0) setIsBuying(true);
+    if (data.minSupply === 0) setIsBuying(false);
+
+    let queryString = `commodityDisplayName=${data.commodityDisplayName.value
+      .split(' ')
+      .join('%20')}&maxLandingPadSize=${data.maxLandingPadSize}&minSupply=${
+      data.minSupply
+    }&minDemand=${data.minDemand}&includeFleetCarriers=${
+      data.includeFleetCarriers
+    }&includeOdyssey=${data.includeOdyssey}&includePlanetary=${
+      data.includePlanetary
+    }&x=0&y=0&z=0`;
 
-    let submitData: ReqBody = {
-      ...data,
-      commodityId: formatString(data.commodityId.value),
-      minDemand: Number(data.minDemand),
-      minSupply: Number(data.minSupply),
-      referenceLocation: {
-        xcoordinate: 0,
-        ycoordinate: 0,
-        zcoordinate: 0,
-      },
-    };
-    delete submitData.system;
+    try {
+      const commodityReq = await fetch(
```

Comment:
Is it worth considering using something like https://www.npmjs.com/package/react-query to manage API requests?

The built in caching, state management and query retries are super useful and allows you to manage api hooks centrally. I have some examples of how Ive used it in react/react native projects I can share to get it set up. 

Maybe worth discussing post-PR?